### ISSUE TITLE
Special processing hooks

### ISSUE THREAD

**Issue by bedeho** [Joystream/hydra#193]
**Note: This is not specifically for Sumer.**

We want to support a more general block explorer like experience, which means we want to connect our Joystream entities to generic entities for blocks, extrinsics, etc. To do this we would like to have ability to specify in the manifest file to run a mapping

- **Block pre-handler:** at the start of each block before any other mappings.
- **Block post-handler:** at the end of each block after all other mappings.
- **Extrinsic pre-handler:** before each and every individual extrinsic and before any other mapping for this extrinsic.
- **Extrinsic post-handler:** after each and every individual extrinsic and after any other mapping for this extrinsic.
- **Extrinsic pre-handler (only for extrinsics of interest):** same as above
- **Extrinsic post-handler (only for extrinsics of interest):** same as above

Obviously the type arguments in question for all are the generic substrate types.

This is the kind of input schema we are looking to safely power simultanosly with other normal joystream concepts

```graphql

type Block @entity {

  "Network in which block occurred."
  network: Network!

  "Height at which block is committed."
  height: BigInteger!

  "Timestamp in block."
  timestamp: BigInteger!

  "Hash of block in hex encoding."
  hash: String!

  "Hash of parent block in hex encoding."
  parentHash: String!

  "State root hash in hex encoding."
  stateRoot: String!

  "Extrinsics merkle root in hex encoding."
  extrinsicRoot: String!

  "Time since last block."
  blockTime: BigInteger!

  # reverse lookup anything that can happeni n a block!
  extrinsics: [Extrinsic!] @derivedFrom(field: "inBlock")

  # All events triggered in block.
  # Not sure if legal.
  #events: [Event!] @derivedFrom(field: "inBlock")

}

# Substrate Root origin type.
type RootOrigin @variant {

  /* empty */
}

# Substrate signed origin type.
type SignedOrigin @variant {

  "Signing account in SS58 encoding."
  signingAccount: String!
}

# Substrate none origin type.
type NoneOrigin @variant {

  /* empty */
}

# Substrate extrinsic origin.
union ExtrinsicOrigin = RootOrigin | SignedOrigin | NoneOrigin

type Extrinsic @entity {

  "Name of runtime module to which extrinsic was targeted."
  moduleName: String!

  "Name of extrinsic."
  call: String!

  "Nonce"
  Nonce: BigInteger!

  "Fee charged."
  fee: BigInteger!

  "Origin for extrinsic."
  origin: ExtrinsicOrigin!

  "Call parameters, unknown encoding currently."
  parameters: String

  "Index of extrinsic in block."
  indexInBlock: UInt!

  "Block in which this extrinsic was included."
  inBlock: Block!

  "Whether extrinsic was successful."
  successful: Boolean!

}

```

**Comment by bedeho** [comment]
The feasibility of this depends on https://github.com/Joystream/hydra/issues/205 to be practical.

**Comment by dzhelezov** [comment]
Block-level hooks are implemented in v3 (#355)

**Comment by dzhelezov** [comment]
The core features are implemented in https://github.com/Joystream/hydra/pull/386 (Hydra v3). Closing for now
